src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/FormTests.cs

```diff
@@ -0,0 +1,141 @@
+                await Task.Delay(SnapLayoutDelayMS);
+
+                // Snap left
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Keyboard.KeyPress(VirtualKeyCode.RIGHT).KeyPress(VirtualKeyCode.RETURN));
+
+                await Task.Delay(SnapLayoutDelayMS);
+
+                // At this point, Windows displays a panel containing all running applications so the
+                // user can select one to dock next to our form. It also takes the keyboard focus away.
+                // If left in this state, subsequently run tests will fail since keyboard focus is not
+                // given to any newly launched window until this panel is dismissed.
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Keyboard.KeyPress(VirtualKeyCode.ESCAPE));
+
+                await Task.Delay(SnapLayoutDelayMS);
+
+                var screenWorkingArea = Screen.FromControl(form).WorkingArea;
+                int borderSize = (form.Width - form.ClientRectangle.Width) / 2;
+
+                Assert.True(form.Left <= screenWorkingArea.X);
+                Assert.True(form.Left >= screenWorkingArea.X - borderSize);
+
+                Assert.True(form.Top <= screenWorkingArea.Y);
+                Assert.True(form.Top >= screenWorkingArea.Y - borderSize);
+
+                Assert.True(form.Height >= screenWorkingArea.Height);
+                Assert.True(form.Height <= screenWorkingArea.Height + (borderSize * 2));
+
+                Assert.True(form.Width >= screenWorkingArea.Width / 2);
+                Assert.True(form.Width <= (screenWorkingArea.Width / 2) + (borderSize * 2));
+            });
+        }
+
+        [WinFormsTheory]
+        [InlineData(FormWindowState.Normal)]
+        [InlineData(FormWindowState.Maximized)]
+        public async Task Form_SnapsRightAsync(FormWindowState windowState)
+        {
+            if (!OsVersion.IsWindows11_OrGreater)
+            {
+                return;
+            }
+
+            await RunEmptyFormTestAsync(async form =>
+            {
+                form.Location = new Point(20, 21);
+                form.Size = new Size(300, 310);
+
+                form.WindowState = windowState;
+
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Keyboard.ModifiedKeyStroke(VirtualKeyCode.LWIN, VirtualKeyCode.VK_Z));
+
+                // inputSimulator.Sleep appears wildly inconsistent with snap panel timing. Task.Delay does not
+                await Task.Delay(SnapLayoutDelayMS);
+
+                // Snap right
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Keyboard.KeyPress(VirtualKeyCode.RIGHT).KeyPress(VirtualKeyCode.RIGHT).KeyPress(VirtualKeyCode.RETURN));
```

Comment:
Makes it easier to read
```suggestion
                    inputSimulator => inputSimulator.Keyboard.KeyPress(VirtualKeyCode.RIGHT)
                                                             .KeyPress(VirtualKeyCode.RIGHT)
                                                             .KeyPress(VirtualKeyCode.RETURN));
```

ditto in other places